### PR TITLE
Feature avl

### PR DESCRIPTION
**Pull request by mclarke2**
# improved user handling of default settings
# AVL induced drag not being used

### CODE REVIEW

File: trunk/SUAVE/Analyses/Aerodynamics/AVL.py [hunk 1]
```diff
@@ -116,8 +121,10 @@ def initialize(self):
         self.geometry
         """          
         self.process.compute.lift.inviscid.geometry = self.geometry
+        sv = self.settings.spanwise_vortices
```
Comment (planes) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL.py]: Unpack these up top

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py [hunk 2]
```diff
@@ -104,7 +101,7 @@ def __defaults__(self):
         # Regression Status
         self.regression_flag                 = False
 
-    def initialize(self):
+    def initialize(self,spanwise_vortices,chordwise_vortices):
```
Comment (planes) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: you need to default the Nones

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py [hunk 3]
```diff
@@ -169,20 +178,28 @@ def evaluate(self,state,settings,geometry):
...
         conditions.aerodynamics.lift_breakdown.inviscid_wings_lift       = Data()    
         conditions.aerodynamics.lift_breakdown.inviscid_wings_lift       = inviscid_lift
-
         state.conditions.aerodynamics.lift_coefficient                   = inviscid_lift
         state.conditions.aerodynamics.lift_breakdown.compressible_wings  = inviscid_lift
         
-        # Inviscid drag, zeros are a placeholder for possible future implementation
-        inviscid_drag                                                    = np.zeros([data_len,1])        
+        # Store inviscid drag results  
+        e             = settings.oswald_efficiency_factor
```
Comment (planes) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: this is span efficiency

File: trunk/SUAVE/Analyses/Stability/AVL.py [hunk 4]
```diff
@@ -159,7 +159,18 @@ def finalize(self):
             fuel.mass_properties.mass  = 0.
             configuration.fuel         = fuel	
 
-
+        # check if user specifies number of spanwise vortices
```
Comment (planes) [inline trunk/SUAVE/Analyses/Stability/AVL.py]: is this the best way to do defaults?

File: trunk/SUAVE/Analyses/Stability/AVL.py [hunk 5]
```diff
@@ -474,18 +485,9 @@ def evaluate_conditions(self,run_conditions):
         batch_template                   = self.settings.filenames.batch_template
         deck_template                    = self.settings.filenames.deck_template
         
-        # check if user specifies number of spanwise vortices
-        if self.settings.spanwise_vortices == None: 
-            spanwise_elements  = self.settings.discretization.defaults.wing.spanwise_elements
-        else:
-            spanwise_elements  = self.settings.spanwise_vortices
+        # rename defaul avl aircraft tag
```
Comment (planes) [inline trunk/SUAVE/Analyses/Stability/AVL.py]: typo

File: trunk/SUAVE/Analyses/Stability/AVL.py [hunk 6]
```diff
@@ -517,7 +519,7 @@ def evaluate_conditions(self,run_conditions):
 
         # write the input files
         with redirect.folder(run_folder,force=False):
-            write_geometry(self,spanwise_elements,chordwise_elements)
+            write_geometry(self)
```
Comment (planes) [inline trunk/SUAVE/Analyses/Stability/AVL.py]: this is inconsistent with the one before

File: trunk/SUAVE/Analyses/Aerodynamics/AVL.py [hunk 7]
```diff
@@ -85,7 +90,7 @@ def __defaults__(self):
         compute.drag.parasite.propulsors.propulsor = Common.Drag.parasite_drag_propulsor
         compute.drag.parasite.pylons               = Common.Drag.parasite_drag_pylon
         compute.drag.parasite.total                = Common.Drag.parasite_total
-        compute.drag.induced                       = Common.Drag.induced_drag_aircraft
+        compute.drag.induced                       = SUAVE.Methods.skip
```
Comment (planes) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL.py]: You could just remove this line

File: trunk/SUAVE/Analyses/Aerodynamics/AVL.py (hunk 1 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL.py]: Moved to the top and inserted #unpack comment

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 6 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: I renamed it spanwise_vortices and chordwise_vortices so that it is consistent with other files in the code. I also appended it to self so you don't need to pass more than one argument into the write_geometry() function

File: trunk/SUAVE/Analyses/Aerodynamics/AVL.py (hunk 7 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL.py]: Removed

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 4 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: currently, yes.  I tried to directly edit the discretization and skip the immediate step but the way AVL was first implemented does not allow it. AVL settings overwrite SUAVE's

File: regression/scripts/Vehicles/Boeing_737.py [hunk 8]
```diff
@@ -30,7 +30,7 @@ def vehicle_setup():
     
     vehicle = SUAVE.Vehicle()
     vehicle.tag = 'Boeing_737800'    
-    
+    vehicle.configuration = 'Tube_Wing'
```
Comment (planes) [inline regression/scripts/Vehicles/Boeing_737.py]: This is not a minor change to SUAVE. This is kind of change requires a group discussion as this changes the ideology behind the code. We're going to need to need to discuss this next meeting if you want to make this type of change.

If you want to finish this pull request quickly I suggest you do this separately.

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py [hunk 9]
```diff
@@ -165,24 +178,35 @@ def evaluate(self,state,settings,geometry):
...
         conditions.aerodynamics.lift_breakdown.inviscid_wings_lift       = Data()    
         conditions.aerodynamics.lift_breakdown.inviscid_wings_lift       = inviscid_lift
-
         state.conditions.aerodynamics.lift_coefficient                   = inviscid_lift
         state.conditions.aerodynamics.lift_breakdown.compressible_wings  = inviscid_lift
         
-        # Inviscid drag, zeros are a placeholder for possible future implementation
-        inviscid_drag                                                    = np.zeros([data_len,1])        
+        # Store inviscid drag results  
+        ar            = geometry.wings['main_wing'].aspect_ratio
```
Comment (planes) [inline trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: why is there a random ar here?

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 6 above)
Comment (planes) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: does this work if you use SU2 aero mixed with AVL stability?

File: trunk/SUAVE/Analyses/Stability/Stability.py [hunk 10]
```diff
@@ -83,8 +83,8 @@ def finalize(self):
 
         Properties Used:
         N/A
-        """          
-
+        """
```
Comment (planes) [inline trunk/SUAVE/Analyses/Stability/Stability.py]: unnecessary change

File: trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py [hunk 11]
```diff
@@ -82,7 +82,7 @@ def translate_avl_geometry(geometry):
                 aircraft.append_wing(w)
                 
         for body in geometry.fuselages:
-                if body.tag == 'fuselage':
+                if body.configuration == 'tube_and_wing' or body.configuration == 'boom':
```
Comment (planes) [inline trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py]: see above comment about configurations

File: trunk/SUAVE/Methods/Center_of_Gravity/compute_fuel_center_of_gravity_longitudinal_range.py [hunk 12]
```diff
@@ -100,9 +100,9 @@ def compute_fuel_center_of_gravity_longitudinal_range(vehicle):
     max_cg      = np.zeros_like(fuel_masses)
     
     tank_masses_front_to_back = tank_masses
-    tank_masses_back_to_front = np.flip(tank_masses)
+    tank_masses_back_to_front = np.flip(tank_masses,0)
```
Comment (planes) [inline trunk/SUAVE/Methods/Center_of_Gravity/compute_fuel_center_of_gravity_longitudinal_range.py]: what is this 0 for?

Comment (mclarke2) [reply trunk/SUAVE/Methods/Center_of_Gravity/compute_fuel_center_of_gravity_longitudinal_range.py]: not sure who/how this was changed, changed it back

File: trunk/SUAVE/Analyses/Stability/AVL.py (hunk 6 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Stability/AVL.py]: that is a good question. since AVL stability runs its own computations I don't see why it won't work. I've run Fidelity Zero Aero with AVL stability already

File: regression/scripts/Vehicles/Boeing_737.py (hunk 8 above)
Comment (mclarke2) [reply regression/scripts/Vehicles/Boeing_737.py]: So I realized this was a mistake. I removed it. The AVL script should read the fuselage configuration NOT vehicle configuration

File: trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py (hunk 11 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py]: I corrected the other file so issues is now resolved

File: trunk/SUAVE/Methods/Center_of_Gravity/compute_fuel_center_of_gravity_longitudinal_range.py (hunk 12 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Center_of_Gravity/compute_fuel_center_of_gravity_longitudinal_range.py]: The 0 is needed when using np.flip as an axis for flipping

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py (hunk 9 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: I was pulling this variable out to compute induced drag later down

File: trunk/SUAVE/Analyses/Stability/Stability.py (hunk 10 above)
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Stability/Stability.py]: I reviewed this file. I don't think any code was changed, just the spaces

Comment (planes) [reply trunk/SUAVE/Analyses/Stability/Stability.py]: I mean:
git checkout develop trunk/SUAVE/Analyses/Stability/Stability.py

File: regression/scripts/Vehicles/Boeing_BWB_450.py [hunk 13]
```diff
@@ -30,8 +30,7 @@ def vehicle_setup():
     # ------------------------------------------------------------------    
 
     vehicle = SUAVE.Vehicle()
-    vehicle.tag = 'Boeing_BWB_450'    
-
+    vehicle.tag = 'Boeing_BWB_450'
```
Comment (planes) [inline regression/scripts/Vehicles/Boeing_BWB_450.py]: why is this file changed?

File: trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py (hunk 9 above)
Comment (planes) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py]: This change is valid in your other pull request.

File: trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py [hunk 14]
```diff
@@ -82,7 +82,7 @@ def translate_avl_geometry(geometry):
                 aircraft.append_wing(w)
                 
         for body in geometry.fuselages:
-                if body.tag == 'fuselage':
+                if body.tag == 'fuselage' or body.tag == 'boom':
```
Comment (planes) [inline trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py]: We need to stop using tag based processes.

If a geometry needs to be ignored or plotted this should be done in the analysis setup.

Ask me for clarification if you're confused.

File: trunk/SUAVE/Methods/Aerodynamics/AVL/write_geometry.py [hunk 15]
```diff
@@ -49,13 +50,13 @@ def write_geometry(avl_object,spanwise_elements,chordwise_elements):
         
         for w in aircraft.wings:
             avl_wing      = translate_avl_wing(w)
-            wing_text     = make_surface_text(avl_wing,spanwise_elements,chordwise_elements)
+            wing_text     = make_surface_text(avl_wing,spanwise_vortices,chordwise_vortices)
             geometry.write(wing_text)  
                      
         for b in aircraft.fuselages:
-            if b.tag == 'fuselage':
+            if b.tag == 'fuselage' or b.tag == 'boom':
```
Comment (planes) [inline trunk/SUAVE/Methods/Aerodynamics/AVL/write_geometry.py]: See above comments about tags

File: trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py [hunk 16]
```diff
@@ -209,7 +209,7 @@ def populate_wing_sections(avl_wing,suave_wing):
                                         segment_tip_chord   = root_chord*suave_wing.Segments[i_segs+1].root_chord_percent
                                         segment_span        = semispan*(suave_wing.Segments[i_segs+1].percent_span_location - suave_wing.Segments[i_segs].percent_span_location )
                                         segment_sweep       = np.arctan(((segment_root_chord*chord_fraction) + (np.tan(sweep_quarter_chord )*segment_span - chord_fraction*segment_tip_chord)) /segment_span)
-                        segment_sweeps.append(segment_sweep)
+                                segment_sweeps.append(segment_sweep)
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py]: Check indentation in this area.

File: regression/scripts/Vehicles/Boeing_BWB_450.py (hunk 13 above)
Comment (mclarke2) [reply regression/scripts/Vehicles/Boeing_BWB_450.py]: Not sure why the changed but I pulled in the develop branch's script

File: trunk/SUAVE/Analyses/Aerodynamics/AVL.py [hunk 17]
```diff
@@ -114,10 +118,15 @@ def initialize(self):
...
+        """  
+        # unpack
+        sv = self.settings.spanwise_vortices
+        cv = self.settings.chordwise_vortices 
+        
         self.process.compute.lift.inviscid.geometry = self.geometry
+        
         # Generate the surrogate
-        self.process.compute.lift.inviscid.initialize()
+        self.process.compute.lift.inviscid.initialize(sv,cv),
```
[the remark this answers is not in the document]
Comment (mclarke2) [reply trunk/SUAVE/Analyses/Aerodynamics/AVL.py]: ",' removed
